File: backend/app/schemas/question_answer/visual_search.py
from typing import List, Dict, Any
from app.schemas.base import BaseSchema
from pydantic import SkipValidation

from app.schemas.question_answer.schema import ImageDocument


# Visual Question Answer
class RequestVisualSearch(BaseSchema):
    question_image: str
    llm: str


class ResponseVisualSearch(BaseSchema):
    image_documents: SkipValidation[List[ImageDocument]]

File: frontend/app/api/question-answer/visual-search.ts
import {backendClient} from "@/app/api/backend";
import {ImageDocument} from "@/app/api/question-answer/index";
import {Response} from "@/app/api";

export interface RequestVisualSearch {
	question_image: string
	llm: string
}

export interface ResponseVisualSearch extends Response {
	image_documents: ImageDocument[]
}

export const ActionVisualSearch = async (
	params: RequestVisualSearch
): Promise<ResponseVisualSearch> => {
	// const endpoint = `/api/system/test/timeout?timeout=3`
	// const res = await backendClient.backend_get(endpoint, params);
	const endpoint = `/api/question_answer/visual_search`;
	const res = await backendClient.backend_post(endpoint, params);

	return res as ResponseVisualSearch;
}